test/math/Polynomial.cpp

```diff
@@ -0,0 +1,164 @@
+#include "math/Polynomial.h"
+#include "BoostUnitTest.h"
+#include <iostream>
+
+BOOST_AUTO_TEST_CASE(LegendreRoots) {
+    std::vector< std::vector <double> > legendreRootsKnown{
+        //order 1
+        {
+            0.0
+        },
+        //order 2
+        {
+            -0.57735026918962584,
+            0.57735026918962562
+        },
+        //order 3
+        {
+            -0.7745966692414834,
+            0.0,
+            0.77459666924148352
+        },
+        //order 4
+        {
+            -0.86113631159405335,
+            -0.3399810435848562,
+            0.33998104358485665,
+            0.86113631159405257
+        },
+        //order 5
+        {
+            -0.9061798459386643,
+            -0.53846931010568333,
+            0.0,
+            0.53846931010568311,
+            0.90617984593866352
+        },
+        //order 6
+        {
+            -0.93246951420315138,
+            -0.66120938646626437,
+            -0.23861918608319704,
+            0.23861918608319688,
+            0.66120938646626337,
+            0.93246951420315194
+        },
+        //order 7
+        {
+            -0.94910791234275782,
+            -0.74153118559939579,
+            -0.40584515137739685,
+            0.0,
+            0.40584515137739685,
+            0.74153118559939546,
+            0.94910791234275971
+        },
+        //order 8
+        {
+            -0.9602898564975284,
+            -0.79666647741363394,
+            -0.52553240991632688,
+            -0.1834346424956497,
+            0.18343464249564989,
+            0.52553240991632955,
+            0.79666647741362762,
+            0.96028985649753373
+        },
+        //order 9
+        {
+            -0.9681602395076313,
+            -0.83603110732662467,
+            -0.6133714327005938,
+            -0.32425342340380914,
+            0.0,
+            0.32425342340380908,
+            0.61337143270059147,
+            0.83603110732663521,
+            0.96816023950762675
+        }
+    };
+
+  int maxOrder = 9;
+
+  for (int order = 1; order <= maxOrder; order++) {
+    std::vector<double> roots = NuTo::Math::Polynomial::LegendreRoots(order);
+    for (int i=0; i<order; i++) {
```

Comment:
This is a problem with values very close to zero, since BOOST_CHECK_CLOSE calculates the relative error, possibly dividing by zero